subsys/net/ip/net_context.c

```diff
@@ -901,6 +878,64 @@ static int create_udp_packet(struct net_context *context,
 }
 #endif /* CONFIG_NET_UDP */
 
+int net_context_output(struct net_context *context, struct net_pkt *pkt,
+		       const struct sockaddr *dst_addr)
+{
+	int ret;
```

Comment:
There needs to be initial value for ret here, perhaps -EPROTONOSUPPORT so that we are not trying to return uninitialized value if UDP or TCP is not configured but proto is set to be UDP or TCP. In this case also the default branch could be removed if the ret is init in the beginning of the func.

Reply:
Oh yes, #if defined(CONFIG_NET_UDP) needs to include all of the case statement, otherwise a request for UDP while UDP is not configured will fall through the cracks.